go.mod

```diff
@@ -152,14 +152,15 @@ replace (
 	dmitri.shuralyov.com/gpu/mtl => dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9
 	github.com/Azure/azure-sdk-for-go => github.com/Azure/azure-sdk-for-go v43.0.0+incompatible
 	github.com/Azure/go-ansiterm => github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
-	github.com/Azure/go-autorest/autorest => github.com/Azure/go-autorest/autorest v0.9.6
-	github.com/Azure/go-autorest/autorest/adal => github.com/Azure/go-autorest/autorest/adal v0.8.2
-	github.com/Azure/go-autorest/autorest/date => github.com/Azure/go-autorest/autorest/date v0.2.0
-	github.com/Azure/go-autorest/autorest/mocks => github.com/Azure/go-autorest/autorest/mocks v0.3.0
+	github.com/Azure/go-autorest => github.com/Azure/go-autorest v14.2.0+incompatible
```

Comment:
This loopback dependency (where a nested module depends on a parent component that is not a module with a module-incompatible version tag) is ... interesting. I'm not sure what issues this could cause in the future.

Looking closer, I can't actually figure out why this dependency was added... there's no code in the base repo, and all subpackages declare their own modules... seems like this could just be dropped as a dependency by the `github.com/Azure/go-autorest/*` modules

Reply:
weird, looks like this was added in https://github.com/Azure/go-autorest/pull/528 ... I'm still not sure why having all the submodules reference a parent containing no code is desired